packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -608,16 +610,29 @@ ProductList.getProps = async ({res, params, location, api}) => {
         res.set('Cache-Control', `max-age=${MAX_CACHE_AGE}`)
     }
 
-    const [category, productSearchResult] = await Promise.all([
-        isSearch
-            ? Promise.resolve()
-            : api.shopperProducts.getCategory({
-                  parameters: {id: categoryId, levels: 0}
-              }),
-        api.shopperSearch.productSearch({
-            parameters: searchParams
-        })
-    ])
+    let category, productSearchResult
+    if (!cachedCategory) {
+        const [freshCategory, freshProductSearchResult] = await Promise.all([
+            isSearch
+                ? Promise.resolve()
+                : api.shopperProducts.getCategory({
+                      parameters: {id: categoryId, levels: 0}
+                  }),
+            api.shopperSearch.productSearch({
+                parameters: searchParams
+            })
+        ])
+        category = freshCategory
+        productSearchResult = freshProductSearchResult
+    } else {
+        category = cachedCategory
+        const [freshProductSearchResult] = await Promise.all([
+            api.shopperSearch.productSearch({
+                parameters: searchParams
+            })
+        ])
+        productSearchResult = freshProductSearchResult
+    }
```

Comment:
instead of this, let's do this:
```
let category, productSearchResult
        const [category,  productSearchResult] = await Promise.all([
            isSearch
                ? Promise.resolve()
                : categories?.[params?.categoryId] 
                ? Promise.resolve(categories?.[params?.categoryId])
                :  api.shopperProducts.getCategory({
                      parameters: {id: categoryId, levels: 0}
                  }),
            api.shopperSearch.productSearch({
                parameters: searchParams
            })
        ])
```